### DISCUSSION TITLE
Documentation on using existing osm2pgsql database missing

### DISCUSSION THREAD

**Discussion by koebi** [osm-search/Nominatim#3399] [Q&A] [answered]
Hey,

we're currently looking into switching our geocoder from Pelias to Nominatim.
Since a big part of the disk space used by Nominatim is the osm2pgsql database, we were wondering if it is possible to build Nominatims indices on an existing database.

If so, how would one go about this? There is unfortunately no documentation regarding this, or at least none I could find.

Best regards,
Jakob

**Comment by mtmail** [comment]
Is the question if you can use a tile-server or overpass database? Nominatim requires its own database. It uses different table structure, it imports different data.

You can have multiple databases on the same server but there will be duplicate data I'm afraid.

**Comment by koebi** [comment]
Nominatim states that it
> uses [osm2pgsql](https://osm2pgsql.org/) to load the OSM data into the PostgreSQL database

This suggests that it should be possible to use an existing database created by something like 
```
osm2pgsql -c my-osm-extract.pbf
```

The docs keep quiet about this, though…

**Reply by mtmail** [reply, marked as answer]
osm2pgsql is a generic tool which uses configuration to set which input goes into which database tables. The default is a `default.style` file (https://osm2pgsql.org/doc/man/version-1-11-0.html). Nominatim uses the flex output (https://osm2pgsql.org/doc/manual.html#the-flex-output) and *.lua scripts (https://github.com/osm-search/Nominatim/tree/master/settings) for its osm2psql configuration. The resulting database tables have different column and contain different data than a default osm2pgql import.

**Comment by lonvia** [comment]
If you feel like experimenting, you can try out https://nominatim.org/tutorials/running-nomintim-and-rendering-together.html

Don't expect to save much space. The only part of the database that is shared by rendering and geocoding is the backstorage. Because of this feedback form people was mixed regarding the question if the saved space is worth the additional setup effort.

**Comment by koebi** [comment]
Thank you both so much for your explanation :)
